docs/core/compatibility/jit/9.0/fp-to-integer.md

```diff
@@ -0,0 +1,72 @@
+
+| Convert to ...            | Value of `x`                           | (Previous) result         |
+|---------------------------|----------------------------------------|---------------------------|
+| `int` scalar and packed   | `int.MinValue <= x <= int.MaxValue`    | `(int)x`                  |
+|                           | `< int.MinValue` or `> int.MaxValue`   | `int.MinValue`            |
+| `long` scalar and packed  | `long.MinValue <= x <= long.MaxValue`  | `(long)x`                 |
+|                           | `< long.MinValue` or `> long.MaxValue` | `long.MinValue`           |
+| `uint` scalar and packed  | Any value                              | `(((long)x << 32) >> 32)` |
+| `ulong` scalar and packed | `<= 2^63`                              | `(long)x`                 |
+|                           | `> 2^63`                               | `(long)(x - 2^63) + 2^63` |
+
+## New behavior
+
+The following table shows the new behavior when converting a `float` or `double` value.
+
+| Convert to ...            | Value of `x`                          | .NET 9+ result   |
+|---------------------------|---------------------------------------|------------------|
+| `int` scalar and packed   | `int.MinValue <= x <= int.MaxValue`   | `(int)x`         |
+|                           | `< int.MinValue`                      | `int.MinValue`   |
+|                           | `> int.MaxValue`                      | `int.MaxValue`   |
+|                           | `NaN`                                 | 0                |
+| `long` scalar and packed  | `long.MinValue <= x <= long.MaxValue` | `(long)x`        |
+|                           | `< long.MinValue`                     | `long.MinValue`  |
+|                           | `> long.MaxValue`                     | `long.MaxValue`  |
+|                           | `NaN`                                 | 0                |
+| `uint` scalar and packed  | `0 <= x <= uint.MaxValue`             | `(uint)x`        |
+|                           | `x > uint.MaxValue`                   | `uint.MaxValue`  |
+|                           | `x < 0`                               | 0                |
+| `ulong` scalar and packed | `0 <= x <= ulong.MaxValue`            | `(ulong)x`       |
+|                           | `x > ulong.MaxValue`                  | `ulong.MaxValue` |
+|                           | `x < 0`                               | 0                |
+
+## Version introduced
+
+.NET 9 Preview 4
+
+## Type of breaking change
+
+This change is a [behavioral change](../../categories.md#behavioral-change).
+
+## Reason for change
+
+This change was made to standardize all floating point-to-integer conversions to have saturating behavior.
+
+## Recommended action
+
+If you relied on any value to be returned from the conversion, even if it was incorrect, update your code to expect the saturated value instead.
```

Comment:
We've also exposed `ConvertToIntegerNative<TInteger> where TIntger : IBinaryInteger<TInteger>` APIs on `Single` (float), `Double` (double), and `Half`.

These APIs use `platform specific behavior`. This is notably not strictly guaranteed to match the prior behavior (which was already non-deterministic), but rather to do whatever is most efficient for the native platform. In most cases, this will be the prior behavior.